e2e/testsuite/testsuite.go

```diff
@@ -317,31 +431,9 @@ func (s *E2ETestSuite) GetChains(chainOpts ...ChainOptionConfiguration) (ibc.Cha
 // GetAllChains returns all chains that can be used in a test. The chains returned
 // are unique to the current test being run. Note: this function does not create containers.
 func (s *E2ETestSuite) GetAllChains(chainOpts ...ChainOptionConfiguration) []ibc.Chain {
-	if s.paths == nil {
-		s.paths = map[string][]ibc.Chain{}
-	}
-
-	if chains, ok := s.paths[s.T().Name()]; ok {
-		return chains
-	}
-
-	chainOptions := DefaultChainOptions()
-	for _, opt := range chainOpts {
-		opt(&chainOptions)
-	}
-
-	chains := s.createChains(chainOptions)
-	s.paths[s.T().Name()] = chains
-
-	if s.proposalIDs == nil {
-		s.proposalIDs = map[string]uint64{}
-	}
-
-	// initialise proposal ids for all chains.
-	for _, chain := range chains {
-		s.proposalIDs[chain.Config().ChainID] = 1
-	}
-
+	// chains are stored on a per test suite level
+	chains := s.chains
+	s.Require().NotEmpty(chains, "chains not found for test %s", s.testSuiteName)
```

Comment:
Maybe not so important, but shouldn't this be up to the test to check if this is important? If they run a standard setup the chains should be there, no? Or is there no scenario where this would actually be used without there being chains? If so, i guess this is fine.

Reply:
this fn is more of a utility function to access the chains (which are created in `SetupSuite` ), IMO I don't think we need to worry about an E2E test that has no chains, I expect this to only fail if there is a test suite overriding `SetupSuite` incorrectly (by not creating chains as part of the setup)

Reply:
Yeah, that makes sense!